### PR TITLE
[GeoMechanicsApplication] Investigate the occasional failure of the "test_parameter_field_with_python_umat_parameters" test on the build server 2nd

### LINKED ISSUE #12711: [GeoMechanicsApplication] Investigate the occasional failure of the "test_parameter_field_with_python_umat_parameters" test on the build server
The TeamCity build server occasionally fails during the _"test_parameter_field_with_python_umat_parameters"_ test.
The error message shows:
`ModuleNotFoundError: No module named 'KratosMultiphysics.GeoMechanicsApplication.user_defined_scripts.custom_field_umat_parameters'.`

This failure happens randomly and needs to be addressed to ensure the test runs successfully and consistently on the build server.

Image

### PR DESCRIPTION
**Pull request by markelov208**
**📝 Description**
The test failed and the following info is available.
 ======================================================================
09:03:34   ERROR: test_parameter_field_with_python_umat_parameters (test_parameter_field.KratosGeoMechanicsParameterFieldTests)
09:03:34   ----------------------------------------------------------------------
09:03:34   Traceback (most recent call last):
09:03:34     File "C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\applications\GeoMechanicsApplication\tests\test_parameter_field.py", line 87, in test_parameter_field_with_python_umat_parameters
09:03:34       simulation = test_helper.run_kratos(file_path)
09:03:34     File "C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\applications\GeoMechanicsApplication\tests\test_helper.py", line 43, in run_kratos
09:03:34       simulation.Run()
09:03:34     File "C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\KratosMultiphysics\analysis_stage.py", line 48, in Run
09:03:34       self.Initialize()
09:03:34     File "C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\KratosMultiphysics\analysis_stage.py", line 92, in Initialize
09:03:34       process.ExecuteInitialize()
09:03:34     File "C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\KratosMultiphysics\GeoMechanicsApplication\set_parameter_field_process.py", line 115, in ExecuteInitialize
09:03:34       custom_module = importlib.import_module("." + self.params["function"].GetString(),
09:03:34     File "C:\python\310\lib\importlib\__init__.py", line 126, in import_module
09:03:34       return _bootstrap._gcd_import(name[level:], package, level)
09:03:34     File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
09:03:34     File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
09:03:34     File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
09:03:34   ModuleNotFoundError: No module named 'KratosMultiphysics.GeoMechanicsApplication.user_defined_scripts.custom_field_umat_parameters'
09:03:34   
09:03:34   Stdout:
09:03:34   Source file path: C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\applications\GeoMechanicsApplication\tests\test_parameter_field\parameter_field_python_umat_parameters\custom_field_umat_parameters.py
09:03:34   Destination file path: C:\BuildAgent\work\6b1b55f40e0ab639\bin\Release\KratosMultiphysics\GeoMechanicsApplication\user_defined_scripts
09:03:34   
It looks that shutil copied the required file to the installation folder. However, the file is not available for importing. 
Therefore, a few more prints where added to check this file in the installation folder.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py [hunk 1]
```diff
@@ -71,21 +73,29 @@ def test_parameter_field_with_python_umat_parameters(self):
         try:
             shutil.copy(custom_python_file, new_custom_script_path)
         except shutil.SameFileError as e:
-             print(f"Source and destination represents the same file. {e}")
-             raise
+            print(f"Source and destination represents the same file. {e}")
+            raise
```
Comment (philbucher) [inline applications/GeoMechanicsApplication/tests/test_parameter_field.py]:
```suggestion
            raise RuntimeError("Source and destination represents the same file") from e
```

This way you get a proper stacktrace (`RuntimeError` can ofc be replaced by sth more appropriate

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py [hunk 2]
```diff
@@ -71,21 +73,29 @@ def test_parameter_field_with_python_umat_parameters(self):
...
-             shutil.copy(custom_python_file, new_custom_script_path)
+            print(f"Try to change the destination file permission {e}")
+            os.chmod(new_custom_script_path, stat.S_IRWXU)
+            shutil.copy(custom_python_file, new_custom_script_path)
         except Exception as e:
-             print(f"Error occurred while copying the file {e}.")
-             raise
+            print(f"Error occurred while copying the file {e}.")
+            raise
+
```
Comment (philbucher) [inline applications/GeoMechanicsApplication/tests/test_parameter_field.py]: you miss `BaseException`

I dont think its likely to occur, but I recommend the empty `except` if you have strange failures

Comment (philbucher) [inline applications/GeoMechanicsApplication/tests/test_parameter_field.py:87]:
```suggestion
        except:
            print("an unknown error occurred")
            raise
```

or if you really wanna catch `BaseException`:
```suggestion
        except BaseException as e:
            raise RuntimeError("an unknown error occurred") from e
```

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py [hunk 3]
```diff
@@ -71,21 +73,29 @@ def test_parameter_field_with_python_umat_parameters(self):
         try:
             shutil.copy(custom_python_file, new_custom_script_path)
         except shutil.SameFileError as e:
-             print(f"Source and destination represents the same file. {e}")
-             raise
+            print(f"Source and destination represents the same file. {e}")
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/test_parameter_field.py]: I think this is one more place we could still use the suggested RuntimeError

**Review by rfaasse — DISMISSED** [review]
Thanks for adding more checks, there is one more place where we can use the RuntimeError suggested by Phil, but other than that, let's merge it and see if we get some more info in the coming days!

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py (hunk 3 above)
Comment (markelov208) [reply applications/GeoMechanicsApplication/tests/test_parameter_field.py]: Thank you, added RuntimeError

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py (hunk 1 above)
Comment (markelov208) [reply applications/GeoMechanicsApplication/tests/test_parameter_field.py]: Thank you for your suggestions. RuntimeError is used now.

File: applications/GeoMechanicsApplication/tests/test_parameter_field.py (hunk 2 above)
Comment (markelov208) [reply applications/GeoMechanicsApplication/tests/test_parameter_field.py]: added `BaseException`

Comment (markelov208) [reply applications/GeoMechanicsApplication/tests/test_parameter_field.py:87]: done the last suggestion.

**Review by markelov208 — COMMENTED** [review]
Hi Philipp and Richard, many thanks for the reviews and suggestions.

**Review by rfaasse — APPROVED** [review]
Thanks for processing the review comments 👍

**Pull request merged**